azkaban-executor/azkaban-emr-jobtype/src/test/java/uk/gov/dwp/dataworks/azkaban/jobtype/EMRStepTest.java

```diff
@@ -0,0 +1,294 @@
+    }
+
+    @Test
+    void shouldNotStartClusterIfJobKilledBeforeInitialAdditionalCheck() {
+        ClusterSummary otherClusterSummary1 = otherClusterSummary(1);
+        ClusterSummary otherClusterSummary2 = otherClusterSummary(2);
+
+        List<ClusterSummary> summaries1 = new ArrayList<>();
+        summaries1.add(otherClusterSummary1);
+        summaries1.add(otherClusterSummary2);
+        ListClustersResult clustersResult1 = mock(ListClustersResult.class);
+
+        when(clustersResult1.getClusters()).thenReturn(summaries1);
+
+        AtomicBoolean killed = new AtomicBoolean(false);
+        AmazonElasticMapReduce emr = mock(AmazonElasticMapReduce.class);
+        when(emr.listClusters(any())).thenAnswer(x -> {
+            killed.set(true);
+            return clustersResult1;
+        });
+
+        AWSLambda lambda = mock(AWSLambda.class);
+        InvokeResult result = mock(InvokeResult.class);
+        when(result.getStatusCode()).thenReturn(200);
+        when(lambda.invoke(any())).thenReturn(result);
+
+        String clusterId = getClusterId(killed, emr, lambda);
+        assertEquals("", clusterId);
+        verify(emr, times(1)).listClusters(any());
+        verifyNoMoreInteractions(emr);
+        verifyNoInteractions(lambda);
+    }
+
+    private String getClusterId(AtomicBoolean killed, AmazonElasticMapReduce emr, AWSLambda lambda) {
+        return EMRStep.getClusterId(emr, lambda, SYSTEM_PROPERTIES, JOB_PROPERTIES, killed,
+                Logger.getLogger(EMRStep.class));
+    }
+
+    @Test
+    void shouldNotStartClusterIfJobKilledAfterInitialAdditionalCheck() {
+        ListClustersResult clustersResult1 = mock(ListClustersResult.class);
+        ListClustersResult clustersResult2 = mock(ListClustersResult.class);
+        ClusterSummary otherClusterSummary1 = otherClusterSummary(1);
+        ClusterSummary otherClusterSummary2 = otherClusterSummary(2);
+
+        List<ClusterSummary> summaries1 = new ArrayList<>();
+        summaries1.add(otherClusterSummary1);
+        summaries1.add(otherClusterSummary2);
+
+        List<ClusterSummary> summaries2 = new ArrayList<>();
+        summaries2.add(otherClusterSummary1);
+        summaries2.add(otherClusterSummary2);
+
+        when(clustersResult1.getClusters()).thenReturn(summaries1);
+        when(clustersResult2.getClusters()).thenReturn(summaries2);
+        AmazonElasticMapReduce emr = mock(AmazonElasticMapReduce.class);
+        AtomicBoolean killed = new AtomicBoolean(false);
+
+        when(emr.listClusters(any())).thenReturn(clustersResult1).thenAnswer(x -> {
+            killed.set(true);
+            return clustersResult2;
+        });
+
+        AWSLambda lambda = mock(AWSLambda.class);
+
+        String clusterId = getClusterId(killed, emr, lambda);
+        assertEquals("", clusterId);
+        verify(emr, times(2)).listClusters(any());
+        verifyNoMoreInteractions(emr);
+        verifyNoInteractions(lambda);
+    }
+
```

Comment:
As mentioned above, can we have a test for cluster doesnt start up after polling time has expired?